lib/compress/zstd_compress.c

```diff
@@ -1542,6 +1552,12 @@ static size_t ZSTD_resetCCtx_internal(ZSTD_CCtx* zc,
 
     zc->isFirstBlock = 1;
 
+    if (params.cParams.strategy >= ZSTD_btopt && params.cParams.windowLog >= 27 && params.nbWorkers == 0) {
```

Comment:
I don't like modifying parameters inside of `ZSTD_resetCCtx_internal()`. At this point they should be fixed.